app/controllers/v0/sessions_controller.rb

```diff
@@ -84,7 +84,12 @@ def saml_callback
 
       if @saml_response.is_valid? && persist_session_and_user
         async_create_evss_account(@current_user)
-        redirect_to Settings.saml.relay + '?token=' + @session.token
+
+        if current_user.loa[:current] < current_user.loa[:highest]
+          redirect_to build_url(authn_context: LOA::MAPPING.invert[3], connect: @current_user&.authn_context)
+        else
+          redirect_to Settings.saml.relay + '?token=' + @session.token
+        end
```

Comment:
Assignment branch condition size is too high just means that we're doing to much branching in this method.

change it to:
`redirect_to saml_callback_success_url`

and move the logic for determining the url to return into a private method down below and you should be fine.